src/System.Private.CoreLib/shared/System/Environment.cs

```diff
@@ -108,12 +107,22 @@ public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption opt
             return GetFolderPathCore(folder, option);
         }
 
+        public static int CurrentManagedThreadId => Thread.CurrentThread.ManagedThreadId;
```

Comment:
This will break corert.

Reply:
It should not, it's used elsewhere too

Reply:
It has a different implementation in corert, which is why this wasn't in shared: coreclr had one implementation, corert has another.
https://github.com/dotnet/corert/pull/6920/files#diff-d009a4adbe3aa344aee46ddfbdff6bf1R15
So as is, this change will break corert when it mirrors over and results in two definitions for Environment.CurrentManagedThreadId